src/Tabs.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+import core from 'metal';
+import templates from './Tabs.soy';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+/**
+ * UI Component to navigate through tabbed data.
+ */
+class Tabs extends Component {
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {Object} tab
+	 * @param {number} index
+	 */
+	addTab(tab, index) {
+		if (core.isNumber(index)) {
+			this.tabs.splice(index, 0, tab);
+		} else {
+			this.tabs.push(tab);
+		}
+	}
+
+	/**
+	 * Adds a tab to the tabs array at the specified index if given or
+	 * appends it at the end.
+	 * @param {string} label
+	 * @param {boolean} disabled
+	 * @param {number} index
+	 */
+	addTabByName(label, disabled, index) {
+		if (core.isString(label) && core.isBoolean(disabled)) {
+			let tab = {
+				label: label,
+				disabled: disabled
+			};
+
+			this.addTab(tab, index);
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	created() {
+		this.tab = this.tabs.findIndex(item => (!item.disabled || !core.isDef(item.disabled)) && !this.disabled);
+
+		if (core.isNull(this.type) && !core.isDefAndNotNull(this.elementClasses)) {
```

Comment:
Similar thing here, default value logic should always be either in `value` or `valueFn` in the state property's configuration.

In this case though I think you could just add another type constant for not using a predefined type, like `Tabs.TYPES.NONE`. That way people can set a type if they wish to customize it, or not set any type to start from scratch, and our logic becomes simpler since you can just set the default value to `Tabs.TYPES.TABS`.